service/src/main/java/bio/terra/cbas/dependencies/sam/SamService.java

```diff
@@ -19,6 +26,23 @@ private StatusApi getStatusApi() {
     return new StatusApi(samClient.getApiClient());
   }
 
+  UsersApi getUsersApi(String accessToken) {
+    return new UsersApi(samClient.getApiClient(accessToken));
+  }
+
+  public Optional<String> getUserToken() {
+    // RequestContextHolder exposes the web request in the form of a *thread-bound*
+    // [RequestAttributes](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/context/request/RequestAttributes.html) object.
+    Object token =
+        RequestContextHolder.currentRequestAttributes()
+            .getAttribute(BearerTokenFilter.ATTRIBUTE_NAME_TOKEN, RequestAttributes.SCOPE_REQUEST);
+    return Optional.ofNullable((String) token);
+  }
+
+  public Optional<UserStatusInfo> getSamUser() {
```

Comment:
Noticed during tests: If there's no token, this method won't throw an error, it'll just return empty user object. That feels like... not the expected behavior? No token should presumably be handled similarly to expired token, where an error is thrown (probably 401). @cjllanwarne am I thinking about this correctly?

Reply:
I think I am leaning more towards the exception because we wouldn't be calling getUser from within a method unless we needed an authenticated user to work with - in which case we would want to error out if we don't have one. I don't foresee us having a situation where we want to allow a non-logged in user to call our endpoints (excluding status api), and I actually think the relay listener will reject before it even gets to this point. If we did have a weird case where we wanted to check the user but didn't care if they existed, I think I'd prefer to have to manually catch the exception to signify we're altering the typical behavior.

Reply:
My thought is that we may be able to just replace this method with `getUserStatusInfo` since that will throw whatever error Sam throws when the token is missing instead of us creating our own exception handling/routing.

Reply:
I think the `getUserStatusInfo` or `getSamUser` would most likely be used when fetching the user ID to either log or store it in database. For checking the permissions we might only be using `getUserToken` method.

Reply:
This would probably mean that our controllers would block the requests before even getting to the logging/storing point, since if the user doesn't have a token they presumably won't have any permissions. We can talk more about this this afternoon, I think the Optionals may not be necessary.